cheevos/cheevos.c

```diff
@@ -2902,24 +2902,21 @@ static int cheevos_iterate(coro_t *coro)
 
       MD5_Init(&coro->md5);
 
-      /* if the file contains header, ignore it */
-      if (coro->len - ((coro->len / 0x2000) * 0x2000) == 512)
-          coro->offset = 512;
-      else
-        coro->offset = 0;
+      /* Checks for the existence of a headered SNES file.
+         Unheadered files fall back to GENERIC_MD5. */
+         
+      if (coro->len % 0x2000 != 512)
+      {
+          coro->gameid = 0;
+          CORO_RET();
+      }
 
+      coro->offset = 512;
       coro->count  = 0;
 
       CORO_GOSUB(EVAL_MD5);
-
-      if (coro->count == 0)
```

Comment:
I removed this part because it seemed redundant. I would like to add it back with a comment if it is useful.